packages/messages/src/browser/notifications-manager.ts

```diff
@@ -0,0 +1,285 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject, postConstruct } from 'inversify';
+import { MessageClient, MessageType, Message as PlainMessage, ProgressMessage, ProgressUpdate, CancellationToken } from '@theia/core/lib/common';
+import { deepClone } from '@theia/core/lib/common/objects';
+import { Event, Emitter } from '@theia/core';
+import { Deferred } from '@theia/core/lib/common/promise-util';
+import { Md5 } from 'ts-md5';
+import * as markdownit from 'markdown-it';
+import throttle = require('lodash.throttle');
+import { NotificationPreferences } from './notification-preferences';
+import { ContextKeyService, ContextKey } from '@theia/core/lib/browser/context-key-service';
+import { OpenerService } from '@theia/core/lib/browser';
+import URI from '@theia/core/lib/common/uri';
+
+export const NotificationManager = Symbol('NotificationManager');
+export interface NotificationManager {
+    readonly open: boolean;
+    readonly onUpdate: Event<NotificationManager.UpdateEvent>;
```

Comment:
`onDidChange` or `onUpdated`? It should indicate that internal state is changed, not assume that it should update some UI elements